CppCoreGuidelines.md

```diff
@@ -12839,7 +12839,104 @@ Type violations, weak types (e.g. `void*`s), and low level code (e.g., manipulat
 
 ### <a name="Rper-Comp"></a>Per.11: Move computation from run time to compile time
 
-???
+Write code that does all its computation during compile time. Failing that, write code that only computes the minimum required during run time.
+
+C++ provides mechanisms for compile time computations:
+
+* `constexpr` functions and expressions
+* Template meta programming
+* Constant values such as enumerations
+* The static type system (See [Per.10: Rely on the static type system](#Rper-type))
+
+#### Reason
+
+Every computation done at compile time won't have to be done at run time. Instead, the results of the compile time computations are already stored into the assembly.
+Having the result of a computation (instead of having to re-calculate it) results in faster execution.
+
+Furthermore, the more you do at compile time, the more the compiler can reason about, and the more the optimizer can optimize.
+If the compiler can determine at compile time that a computation won't ever be run, it can be removed.
+
+This effect can cascade: the more that the compiler can optimize, the more it might be able to optimize further.
+In particular, any branches that can be removed from performance-critical code will remove the run time cost of branch mis-prediction.
+If all the branches in a function have conditions with values known at compile time, the entire function will be reduced to a branch-less block of statements and expressions. Such functions are great for reducing cache pressure.
+
+#### Example
+
+Here's a `constexpr` function for calculating Fibonacci Numbers:
+
+    static constexpr int fib(int const n)
+    {
+        return (n <= 0) ? 0 :
+                  (n == 1) ? 1 :
+                     fib(n-1) + fib(n-2);
+    }
+
```

Comment:
Thanks for the change, but what is the purpose of `ith_minus_one`? And you are missing a return.